plugins/optimization-detective/optimization.php

```diff
@@ -245,8 +245,10 @@ function od_optimize_template_output_buffer( string $buffer ): string {
 		return $buffer;
 	}
 
-	$slug = od_get_url_metrics_slug( od_get_normalized_query_vars() );
-	$post = OD_URL_Metrics_Post_Type::get_post( $slug );
+	$query_vars = od_get_normalized_query_vars();
+	$slug       = od_get_url_metrics_slug( $query_vars );
+	$post       = OD_URL_Metrics_Post_Type::get_post( $slug );
+	$post_id    = $post instanceof WP_Post && $post->ID > 0 ? $post->ID : null;
```

Comment:
I find this a bit strange - I assume without it PHPStan complains. But in reality a `WP_Post` should never have a non-positive `ID`. Obviously an additional check for it is not a real problem, but if we then set the `$post_id` to `null` if not, we should also set the `$post` to `null` in that same situation.

In other words: If somehow the `WP_Post` is invalid, we shouldn't allow the post object to get passed through either, if we don't accept its `ID` value.

Reply:
Yes, it's for the sake of PHPStan. The condition was present before when constructing `OD_Tag_Visitor_Context` but just moved up here.

Technically `WP_Post` can have a negative integer _sometimes_. It was a hack I used in the Customizer to denote `nav_menu_item` posts which were in a changeset but not yet published to the database 😅 

Otherwise, the `$post` is only used in `OD_URL_Metrics_Post_Type::get_url_metrics_from_post()` below, which only looks at the `post_content` of the object, so in that case it doesn't matter if the `ID` is not valid.

Reply:
But if a post doesn't have an ID, wouldn't it be considered irrelevant for Optimization Detective too? I think it would make more sense to then nullify it too for a consistent experience to avoid a weird edge-case.

Reply:
Since in practice it will never not be a `positive-int`, I've opted for phpdoc to declare it as such: 0d8cb6540.